src/support/scratch.c

```diff
@@ -155,6 +155,85 @@ __wt_buf_set_printable(
 	return (buf->data);
 }
 
+/*
+ * __wt_buf_set_printable_format --
+ *	Set the contents of the buffer to a printable representation of a byte
+ * string, based on a format.
+ */
+const char *
+__wt_buf_set_printable_format(WT_SESSION_IMPL *session,
+    const void *buffer, size_t size, const char *format, WT_ITEM *buf)
+{
+	WT_DECL_ITEM(tmp);
+	WT_DECL_PACK_VALUE(pv);
+	WT_DECL_RET;
+	WT_PACK pack;
+	const uint8_t *p, *end;
+	const char *retp, *sep;
+
+	p = (const uint8_t *)buffer;
+	end = p + size;
+
+	WT_ERR(__wt_buf_init(session, buf, 0));
+
+	WT_ERR(__pack_init(session, &pack, format));
+	for (sep = ""; (ret = __pack_next(&pack, &pv)) == 0;) {
+		WT_ERR(__unpack_read(session, &pv, &p, (size_t)(end - p)));
+		switch (pv.type) {
+		case 'x':
+			break;
+		case 's':
+		case 'S':
+			WT_ERR(__wt_buf_catfmt(
+			    session, buf, "%s%s", sep,  pv.u.s));
```

Comment:
Adding quotes around strings might be better so it's possible to know whether a `,` is a separator or part of a string. Though I guess that'll mean strings containing quote characters will be difficult to interpret. Another option might be to print out the string length as part of the output, but that will be difficult to separate from other fields. Maybe keeping it simple is best?

Reply:
@agorrod 

> I went through the exact same thought process, and futzed with it a bit (and we already put `{}` around the key/value pairs, it's not just commas).

I'd be inclined to either leave it alone or insert backslashes in front of `{}` and commas, but don't feel strongly.

Reply:
Cool - this gets an lgtm from me then.

Reply:
@agorrod 

Which do you prefer? Leave alone, or backslash escape?

Reply:
Leave it alone